pkg/testutils/e2e/kubectl/kubectl.go

```diff
@@ -177,3 +185,109 @@ func (c *Client) Apply(ctx context.Context, yaml string) error {
 
 	return nil
 }
+
+type portForwardConfig struct {
+	localport uint
+	stderr    io.Writer
+	stdout    io.Writer
+}
+
+// PortForwardOption defines a configuration option for port forwarding
+type PortForwardOption func(portForwardConfig) portForwardConfig
+
+// WithLocalPort sets the local port to listen for request. Defaults to 0 (random local port)
+func WithLocalPort(port uint) PortForwardOption {
+	return func(p portForwardConfig) portForwardConfig {
+		p.localport = port
+		return p
+	}
+}
+
+// WithOutputStreams sets the output streams for the port forwarder. Default to nil
+func WithOutputStreams(stdout io.Writer, stderr io.Writer) PortForwardOption {
+	return func(p portForwardConfig) portForwardConfig {
+		p.stdout = stdout
+		p.stderr = stderr
+		return p
+	}
+}
+
+func newPortForwardConfig(opts ...PortForwardOption) portForwardConfig {
+	config := portForwardConfig{
+		localport: 0,
+		stdout:    nil,
+		stderr:    nil,
+	}
+
+	for _, option := range opts {
+		config = option(config)
+	}
+
+	return config
+}
+
+// ForwardPodPort opens a local port for forwards requests to a pod's port.
+// Returns the local port used for listening
+func (c *Client) ForwardPodPort(
+	ctx context.Context,
+	pod string,
+	namespace string,
```

Comment:
Having the pod name before the namespace seems a bit odd, I think the k8s libraries, as well as other functions in the project take the namespace first.